src/trajectory.jl

```diff
@@ -587,79 +525,67 @@ function isterminated(h::Hamiltonian, t::BinaryTree{<:ClassicNoUTurn})
 end
 
 """
-    isterminated(h::Hamiltonian, t::BinaryTree{<:GeneralisedNoUTurn})
+    isterminated(h::Hamiltonian, t::BinaryTree)
```

Comment:
```suggestion
    $(SIGNATURES)
```

Similiar for all other places.